Modules/Core/Common/src/itkThreadPool.cxx

```diff
@@ -30,22 +30,28 @@
 namespace itk
 {
 
+std::once_flag globalThreadPoolOnceFlag;
+
 struct ThreadPoolGlobals
 {
   ThreadPoolGlobals() = default;
-  // To lock on the internal variables.
+
+  // To lock on the various internal variables.
   std::mutex          m_Mutex;
+
+  // The singleton instance of ThreadPool.
   ThreadPool::Pointer m_ThreadPoolInstance;
+
 #if defined(_WIN32) && defined(ITKCommon_EXPORTS)
   // ThreadPool's destructor is called during DllMain's DLL_PROCESS_DETACH.
   // Because ITKCommon-5.X.dll is usually being detached due to process termination,
   // lpvReserved is non-NULL meaning that "all threads in the process
   // except the current thread either have exited already or have been
   // explicitly terminated by a call to the ExitProcess function".
   // Therefore we must not wait for the condition_variable.
-  bool m_WaitForThreads = false;
+  bool m_WaitForThreads = false; // guarded by m_PimplGlobals->m_Mutex?
```

Comment:
@dzenanz one thing I couldn't convince myself of... does this bool expect/require the mutex?

Reply:
It does not expect nor require the mutex. It is initialized in a complicated way depending on compile-time conditionals, and we allow user to change it later. But usually, it will not change at run time.

Reply:
OK, I've amended my code: made it std::atomic instead.